packages/ember-routing/lib/services/routing.js

```diff
@@ -0,0 +1,102 @@
+/**
+@module ember
+@submodule ember-routing
+*/
+
+import Service from "ember-runtime/system/service";
+
+import { get } from "ember-metal/property_get";
+import { readOnly } from "ember-metal/computed_macros";
+import { routeArgs } from "ember-routing/utils";
+import keys from "ember-metal/keys";
+import merge from "ember-metal/merge";
+
+/** @private
+  The Routing service is used by LinkView, and provides facilities for
+  the component/view layer to interact with the router.
+
+  While still private, this service can eventually be opened up, and provides
+  the set of API needed for components to control routing without interacting
+  with router internals.
+*/
+
+var RoutingService = Service.extend({
+  router: null,
```

Comment:
Would be much nicer if we could just annotate this with `Ember.inject('router:main')` as it works with inheritance and all that.

also opens the future of using real annotations.

```js
@inject('router', 'router:main');
export default class RoutingService extends Service { 

}
```